src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/spark/rpc/PSRpcObject.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysml.runtime.controlprogram.paramserv.spark.rpc;
+
+import java.nio.ByteBuffer;
+
+import org.apache.sysml.runtime.controlprogram.caching.CacheableData;
+import org.apache.sysml.runtime.instructions.cp.ListObject;
+
+public abstract class PSRpcObject {
+
+	public static final String PUSH = "push";
+	public static final String PULL = "pull";
+	public static final String DATA_KEY = "data";
+	public static final String EMPTY_DATA = "";
+
+	public abstract void deserialize(ByteBuffer buffer);
+
+	public abstract ByteBuffer serialize();
+
+	/**
+	 * Convert direct byte buffer to string
+	 * @param buffer direct byte buffer
+	 * @return string
+	 */
+	protected String bufferToString(ByteBuffer buffer) {
+		byte[] result = new byte[buffer.limit()];
+		buffer.get(result, 0, buffer.limit());
+		return new String(result);
```

Comment:
This can be removed once we modified the serialization/deserialization. Usually it's not a good idea to convert anything other than meta data into string representations as its conversion and parsing is very expensive for floating point data.